File: LeetCode/Medium/0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long ret = 0;

        while (x != 0){
            ret = ret * 10 + x % 10;
            x /= 10;
            
        }
        
        if (ret < Math.pow(-2, 31) || ret > Math.pow(2, 31) - 1) return 0;
    
        return (int) ret;
    }
}

File: LeetCode/Medium/0007-reverse-integer/0007-reverse-integer.go
func reverse(x int) int {
    ret := 0
    
    for x != 0 {
        if ret > 214748364 || ret < -214748364 { return 0 }
        ret = ret * 10 + x % 10
        x /= 10
    }
    
    return ret
}